sbatch_scripts/gen_sbatch.py

```diff
@@ -0,0 +1,406 @@
+    subparsers = parser.add_subparsers(
+        title="commands",
+        dest="command",
+        required=True,
+    )
+    # Generate args
+    generate_parser = subparsers.add_parser(
+        "generate", help="Generates simulated data from model parameters", 
+        parents=[parent_parser_config]
+    )
+
+    # Jaxtrain args
+    jaxtrain_parser = subparsers.add_parser(
+        "jaxtrain", help = "Trains a neural network using simulated data", 
+        parents=[parent_parser_config]
+    )
+    jaxtrain_parser.add_argument(
+        "--training-data-folder",
+        help="Path to folder with data to train the neural network on",
+        type=str,
+        required=True
+    )
+    jaxtrain_parser.add_argument(
+        "--network-id",
+        type=int,
+        help="Id for the neural network to train (default=0).",
+        default=0,
+    )
+    jaxtrain_parser.add_argument(
+        "--dl-workers",
+        type=int,
+        help="Number of cores to use with the dataloader class (default=1)",
+        default=0,
+        required=True
+    )
+
+    # Manually adding metadata parent arguments to the subparsers so the order is correct
+    for action in parent_parser_metadata._actions:
+        if isinstance(action, argparse._HelpAction):
+            continue  # skip help, already there
+        jaxtrain_parser._add_action(action)
+        generate_parser._add_action(action)
+
+    args = parser.parse_args()
+
+    # Setting up logging
+    logger = logging.getLogger(__file__)
+    logging.basicConfig(
+        format="%(asctime)s [%(name)s] %(levelname)s: %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+        level=getattr(logging, args.log_level)
+    )
+
+    if args.command == "generate":
+        
+        # target folder for generate
+        target = Path(args.output_path)
+        target.mkdir(exist_ok=True, parents=True)
+        logger.info(f"Simulated data output folder: {target}")
+
+        # Add correct environment to SBATCH file
+        environment=get_environment_setup(args.make_env, args.command)
+
+        # get parameters for command from the arguments parser
+        params = get_parameters_setup(args.command, args)
+
+        # Create command
+        command = create_command("ssms/cli/generate.py", **params)
+        logger.info(f"Generated command: {command}")
+
+        # Get configuration file metadata 
+        bc = get_basic_config_from_yaml(params["config-path"])
+        
+        # Use information from config file to name job and sbatch script
+        job_name = f"{bc['MODEL']}_generate_sbatch" 
+        script = f"{bc['MODEL']}_generate_sbatch.sh"
+
+        # Create SBATCH metadata
+        sbatch_script = create_sbatch_script(
+                job_name=job_name,
+                output=f"{job_name}.out",
+                error=f"{job_name}.err",
+                time=args.time,
+                command=command,
+                mem="16G",
+                environment=environment
```

Comment:
Update the call to `create_sbatch_script`.
```suggestion
                environment=environment
                array_size=args.array_size,
```